docs/arch/plugins_actions.md

```diff
@@ -0,0 +1,53 @@
+# RDIFF-BACKUP ACTIONS PLUGINS
+
+## Naming convention and actions manager
+
+An action plug-in is:
+
+1. either in the `rdiffbackup.actions` namespace, with any module name (this is
+   how built-in action modules are packaged),
+2. or a top-level module with a name `rdb_action_[...].py`
+
+All those plug-ins are found by the action manager module `rdiffbackup.actions_mgr`
+and returned as a hash made of key-value pairs by the function `get_discovered_actions`:
+
+* key is the name returned by the Action class (see below the `get_name` method)
+* value is the Action class itself, as returned by the `get_action_class` function
+  of the action plug-in module.
+
+> **NOTE:** more information is provided with `pydoc rdiffbackup.actions_mgr`.
+
+## Module interface
```

Comment:
Would this approach allow to integrate rdiff-backup-delete into a plugin ?

Reply:
That's the idea indeed, it was one of the drivers for this plug-in concept :-)

But don't start yet, I expect the interface to need much extension before it can be done properly (as hinted in the docs). It's only a first small step.

Reply:
This said, my idea was to add a sub-action to the remove action, so it would be implemented as `remove files` vs. currently only `remove increments`. It would hence not be really a plug-in but the fact that plug-ins are possible would make it easier to implement properly (so the plan).
But, anyway, a user could implement it as `delete-files` plug-in and it would work as well.

Reply:
Yeah !